api/dispatcher.proto

```diff
@@ -157,3 +165,55 @@ message TasksMessage {
 	repeated Task tasks = 1;
 }
 
+message AssignmentsRequest {
+	string session_id = 1 [(gogoproto.customname) = "SessionID"];
+}
+
+message AssignmentsMessage {
+	// AssignmentType specifies whether this assignment message carries
+	// the full state, or is an update to an existing state.
+	enum AssignmentType {
+		COMPLETE = 0;
+		INCREMENTAL = 1;
+	}
+
+	AssignmentType assignment_type = 1;
+
+	// Sequence is the sequence number of the assignment message. This
+	// starts at 0 and increments. If a sequence number is skipped, the
+	// consumer of the stream should start a new Assignments stream to
+	// re-sync.
+	int64 sequence = 2;
+
+	// UpdateTasks is a set of new or updated tasks to run on this node.
+	// In the first assignments message, it contains all of the tasks
+	// to run on this node. Tasks outside of this set running on the node
+	// should be terminated.
+	repeated Task update_tasks = 3;
```

Comment:
There might be something nice we can do here with `Any` or a `oneof`. I suspect that secrets and tasks are the last thing we add here.